OpenDreamClient/Rendering/ClientImagesSystem.cs

```diff
@@ -69,17 +52,21 @@ internal sealed class ClientImagesSystem : SharedClientImagesSystem {
     private void OnRemoveClientImage(RemoveClientImageEvent e) {
         EntityUid ent = _entityManager.GetEntity(e.AttachedEntity);
         if(ent == EntityUid.Invalid) {
-                if(!_turfClientImages.TryGetValue(e.TurfCoords, out var iconList))
-                    return;
-                iconList.Remove(e.ImageAppearance);
+            if(!_turfClientImages.TryGetValue(e.TurfCoords, out var iconList))
+                return;
+            iconList.Remove(e.ImageEntity);
+            if(iconList.Count == 0)
+                _turfClientImages.Remove(e.TurfCoords);
+            else
                 _turfClientImages[e.TurfCoords] = iconList;
```

Comment:
This should be unnecessary. `_turfClientImages[e.TurfCoords]` is already a reference to `iconList`.